docs/how-to-guides/data-basics/3-building-an-edit-form.md

```diff
@@ -0,0 +1,521 @@
+
+```js
+import { Button, Modal, TextControl } from '@wordpress/components';
+
+function PageEditButton({ pageId }) {
+	const [ isOpen, setOpen ] = useState( false );
+	const openModal = () => setOpen( true );
+	const closeModal = () => setOpen( false );
+	return (
+		<>
+			<Button
+				onClick={ openModal }
+				variant="primary"
+			>
+				Edit
+			</Button>
+			{ isOpen && (
+				<Modal onRequestClose={ closeModal } title="Edit page">
+					<EditPageForm
+						pageId={pageId}
+						onCancel={closeModal}
+						onSaveFinished={closeModal}
+					/>
+				</Modal>
+			) }
+		</>
+	)
+}
+```
+
+When you click the *Edit* button now, you should see the following modal:
+
+![](./media/edit-form/form-scaffold.png)
+
+Great! We now have a basic user interface to work with.
+
+### Step 3: Populate the form with page details
+
+We want the `EditPageForm` to display the title of the currently edited page. You may have noticed that it doesn't receive a `page` prop, only `pageId`. That's okay. Gutenberg Data allows us to easily access entity records from any component.
+
+In this case, we need to use the [`getEntityRecord`](/docs/reference-guides/data/data-core/#getentityrecord) selector. The list of records is already available thanks to the `getEntityRecords` call in `MyFirstApp`, so there won't even be any additional HTTP requests involved – we'll get the cached record right away.
+
+Here's how you can try it in your browser's dev tools:
+
+```js
+wp.data.select( 'core' ).getEntityRecord( 'postType', 'page', 9 );  // Replace 9 with an actual page ID
+```
+
+Let's update `EditPageForm` accordingly:
+
+```js
+export function EditPageForm( { pageId, onCancel, onSaveFinished } ) {
+	const page = useSelect(
+		select => select( coreDataStore ).getEntityRecord( 'postType', 'page', pageId ),
+		[pageId]
+	);
+	return (
+		<div className="my-gutenberg-form">
+			<TextControl
+				label='Page title:'
+				value={ page.title }
+			/>
+			{ /* ... */ }
+		</div>
+	);
+}
+```
+
+Now it should look like that:
+
+![](./media/edit-form/form-populated.png)
+
+### Step 4: Making the Page title field editable
+
+There's one problem with our _Page title_ field: you can't edit it. It receives a fixed `value` but doesn't update it when typing. We need an `onChange` handler.
+
+
+You may have seen a pattern similar to this one in other React apps. It's known as a ["controlled component"](https://reactjs.org/docs/forms.html#controlled-components):
+
+```js
+export function VanillaReactForm({ initialTitle }) {
+	const [title, setTitle] = useState( initialTitle );
+	return (
+		<TextControl
+			value={ title }
+			onChange={ setTitle }
+		/>
+	);
+}
+```
+
+Updating entity records in Gutenberg Data is similar but instead of using `setTitle` to store in local state, we use the `editEntityRecord` action which stores the updates in the _Redux_ state. Here's how you can try it out in your browser's dev tools:
```

Comment:
```suggestion
Updating entity records in Gutenberg Data is similar but instead of using `setTitle` to store in local (component level) state, we use the `editEntityRecord` action which stores the updates in the _Redux_ state. Here's how you can try it out in your browser's dev tools:
```